field/goldilocks/goldilocks.go

```diff
@@ -0,0 +1,191 @@
+package goldilocks
+
+// Partially wraps and extends the functionality of the goldilocks field package.
+
+import (
+	"fmt"
+
+	g "github.com/consensys/gnark-crypto/field/goldilocks"
+)
+
+type Element = g.Element
+
+const Bytes = 8
+
+func reverseBytes(b []byte) []byte {
+	res := make([]byte, len(b))
+	for i, j := 0, len(b)-1; i < j; i, j = i+1, j-1 {
+		res[i], res[j] = b[j], b[i]
+	}
+	return res
+}
+
+func ArrayFromNonCanonicalLittleEndianBytes(in []byte) ([]Element, error) {
+	inClone := make([]byte, len(in))
+	copy(inClone, in)
+
+	// Make up to a multiple of 8 bytes
+	if len(inClone)%8 != 0 {
+		inClone = append(inClone, make([]byte, 8-len(inClone)%8)...)
+	}
```

Comment:
I think creating array with length of `8 * upper[len / 8]` is better, otherwise this append could copy whole array again